services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/utils/GoalStateManager.java

```diff
@@ -32,6 +32,7 @@
 
 @Component
 public class GoalStateManager {
+  public static final int GOAL_STATE_MESSAGE_FORMAT_VERSION = 101;
```

Comment:
sorry what is this about?

Reply:
see end of number 4 of #386. The idea is DPM will stamp a version number (101), and then ACA will look at it to ensure compability, this will enable ACA to print a meaningful error message. And it will save us to trouble trying to figure out incompatability goalstate message.

Reply:
Okay this top-level format version may increase in the future therefore we would likely need to make it configurable. 

Also, GoalStateManager is used by v1.0 so we would need to make a similar change to v2.0.

Reply:
okay, let me know if I should make the configurable change (please give me some pointer to do it).

I made the v2.0 change in DataPlanceServiceImpl.java and updated the PR.

Reply:
Sure. Here are three steps to make the change:
(1) Define a configuration in application.properties with default value
https://github.com/futurewei-cloud/alcor/blob/master/services/data_plane_manager/src/main/resources/application.properties

(2) "Autowire" the configuration to a java config class
https://github.com/futurewei-cloud/alcor/blob/master/services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/config/Config.java

(3) Assign the config value in the code to a local variable, for example, 
https://github.com/futurewei-cloud/alcor/blob/ff19c6b5bdf45b81572409e66201b04fe94a6fcb/services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/client/grpc/DataPlaneClientImpl.java#L49

Reply:
Added for the v2.0 path.